assets/scripts/users/Avatar.jsx

```diff
@@ -1,104 +1,59 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
+import axios from 'axios'
 import { API_URL } from '../app/config'
 import { rememberUserProfile } from '../store/actions/user'
 import './Avatar.scss'
 
-// Requests are cached so that multiple Avatar components that have the
-// same userId only need to make one request. Requests are cached separately
-// from user profile data because this also remembers a failed request (and
-// won't make subsequent requests if it failed previously.)
-const requests = {}
+function Avatar (props) {
+  const { userId, image, rememberUserProfile } = props
 
-export class Avatar extends React.PureComponent {
-  static propTypes = {
-    userId: PropTypes.string.isRequired,
-    profileCache: PropTypes.object,
-    rememberUserProfile: PropTypes.func
-  }
-
-  static defaultProps = {
-    rememberUserProfile: () => {}
-  }
-
-  constructor (props) {
-    super(props)
-
-    this.state = {
-      image: null
-    }
-  }
-
-  async componentDidMount () {
-    // If a profile image had not been cached, initiate a fetch of it
-    let url = this.getCachedProfileImageUrl(this.props.userId)
+  useEffect(() => {
+    async function fetchData () {
+      const response = await axios(API_URL + 'v1/users/' + userId)
 
-    if (!url) {
-      url = await this.fetchAvatar(this.props.userId)
+      // Responses are cached so that multiple Avatar components that
+      // have the same userId only need to make one request.
+      if (response.data) {
+        rememberUserProfile(response.data)
```

Comment:
Woah, this is a pretty awesome use of hooks. Makes the caching process really transparent!